mod/messages/start.php

```diff
@@ -243,18 +240,22 @@ function messages_send($subject, $body, $recipient_guid, $sender_guid = 0, $orig
 		add_entity_relationship($message_sent->guid, "reply", $original_msg_guid);
 	}
 
-	$message_contents = strip_tags($body);
 	if (($recipient_guid != elgg_get_logged_in_user_guid()) && $notify) {
+		$message_contents = $body;
 		$recipient = get_user($recipient_guid);
 		$sender = get_user($sender_guid);
 		
 		$subject = elgg_echo('messages:email:subject', [], $recipient->language);
 		$body = elgg_echo('messages:email:body', [
 				$sender->getDisplayName(),
 				$message_contents,
-				elgg_get_site_url() . "messages/inbox/" . $recipient->username,
+				elgg_generate_url('collection:object:messages:owner', [
+					'username' => $recipient->username,
+				]),
 				$sender->getDisplayName(),
-				elgg_get_site_url() . "messages/add?send_to=" . $sender_guid,
+				elgg_http_add_url_query_elements(elgg_generate_url('add:object:messages'), [
```

Comment:
You can pass query elements to elgg_generate_url(), if they are not part of the route pattern, they will be appended as query elements.

Reply:
i learn something new every day, thanks